modules/hisilicon/hisi.c

```diff
@@ -0,0 +1,74 @@
+/**
+ * @file hisi.c  HiSilicon sound driver
+ *
+ * Copyright (C) 2022 Dmitry Ilyin
+ */
+
+#include <re.h>
+#include <baresip.h>
+
+#include "hisi.h"
+
+#include <mpi_sys.h>
+#include <mpi_vb.h>
+
+static struct ausrc *ausrc;
+static struct auplay *auplay;
+
+static int init_hw(void) {
+	int ret = HI_MPI_SYS_Init();
+
+	if (HI_SUCCESS != ret)
+	{
+		printf("HI_MPI_SYS_Init failed with %d\n", ret);
+		HI_MPI_VB_Exit();
+	}
+	return ret;
+}
+
+unsigned audio_frame_size(unsigned srate) {
+	switch (srate) {
+		case 8000: return 8000 / 50;
+		case 12000: return 12000 / 50;
+		case 16000: return 16000 / 50;
+		case 24000: return 24000 / 50;
+		case 48000: return 48000 / 50;
+		default:
+			    return 320;
+	}
```

Comment:
can be replaced with a simpler formula:

```c
return srate / 50;
```

can you also describe what is 50 and how is it calculated ?